src/device/specification.go

```diff
@@ -0,0 +1,25 @@
+package device
+
+import "fmt"
+
+func (d Device) RequestSpecification(client Client) error {
+	resp, err := client.Specification(d)
+	if err != nil {
+		d.logger.Warn(fmt.Sprintf("Error loading specification for [%s]", d.ID.String()))
+		return err
+	}
+
+	spec, err := client.EvaluateSpecificationResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	d.LoadFromSpecification(spec)
+	return nil
+}
+
+func (d *Device) LoadFromSpecification(spec Device) {
```

Comment:
pointer methods dont get mocked which make it harder to test down the line. Id recommend returning itself is a more flexible move. it also then allows for chaining of functions.

```
# before
func (d *Device) LoadFromSpecification(spec Device) {

# after
func (d Device) LoadFromSpecification(spec Device) Device {